core/src/main/scala/kafka/tools/ConsoleProducer.scala

```diff
@@ -220,11 +228,28 @@ object ConsoleProducer {
     options = tryParse(parser, args)
 
     CommandLineUtils.printHelpAndExitIfNeeded(this, "This tool helps to read data from standard input and publish it to Kafka.")
-    CommandLineUtils.checkRequiredArgs(parser, options, topicOpt, brokerListOpt)
+
```

Comment:
No need to manually check these if you use `.requiredUnless("broker-list")`.

Reply:
Updated with requiredUnless